src/ls_server.rs

```diff
@@ -61,8 +62,8 @@ enum Method {
 
 #[derive(Debug)]
 enum Notification {
-    CancelRequest(usize),
-    Change(ChangeParams),
+    CancelRequest(NumberOrString),
```

Comment:
Yeah, it's an id. (Well, one that is not `null` at least. Tecnically in JSON-RPC `null` is allowed as an id, but discouraged)
But, if we want to use the same type, the other PR would have to use the `languageserver-types` type, not the other way around, because that would require adding a dependency on `rustls` to `languageserver-types`, which would not be suitable.

Anyways, it's likely that `Id` will be replaced eventually, if/when a PR comes to use RustLSP or another JSON-RPC library.